ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,403 @@
+from ansible_collections.arista.avd.plugins.module_utils.eos_designs import AVDFacts
+from functools import cached_property
+from ansible_collections.arista.avd.plugins.module_utils.utils import get
+
+
+class AVDStructConfig(AVDFacts):
+
+    def __init__(self, hostvars=None):
+        super().__init__(hostvars)
+
+    @cached_property
+    def _mgmt_interface_vrf(self):
+        return get(self._hostvars, 'mgmt_interface_vrf')
+
+    @cached_property
+    def _mgmt_gateway(self):
+        return get(self._hostvars, 'mgmt_gateway')
+
+    @cached_property
+    def _switch_data(self):
+        return get(self._hostvars, 'switch')
+
+    @cached_property
+    def _internal_vlan_order(self):
+        return get(self._hostvars, 'internal_vlan_order')
+
+    @cached_property
+    def _range(self):
+        return get(self._internal_vlan_order, 'range')
+
+    @cached_property
+    def _platform_settings(self):
+        return get(self._switch_data, 'platform_settings')
+
+    @cached_property
+    def router_bgp(self):
+        bgp_as = get(self._switch_data, "bgp_as")
+        if bgp_as is not None:
+            router_bgp = {'as': get(self._switch_data, 'bgp_as'),
+                          'router_id': get(self._switch_data, 'router_id'),
+                          'bgp_defaults': []}
+            for bgp_default in get(self._switch_data, 'bgp_defaults'):
+                router_bgp['bgp_defaults'].append(bgp_default)
+
+            if get(self._hostvars, 'bgp_maximum_paths') is not None and get(self._hostvars, 'bgp_ecmp') is not None:
+                router_bgp['bgp_defaults'].append(f"maximum-paths {str(get(self._hostvars, 'bgp_maximum_paths'))} "
+                                                  f"ecmp {str(get(self._hostvars, 'bgp_ecmp'))}")
+            elif get(self._hostvars, 'bgp_maximum_paths') is not None:
+                router_bgp['bgp_defaults'].append(f"maximum-paths {str(get(self._hostvars, 'bgp_maximum_paths'))}")
+            return router_bgp
```

Comment:
Let's try to make the code not a one to one copy of Jinja2 template as we have been using some cumbersome logic in Jinja that can be alleviated by python syntax. E.g.:

```suggestion
    def router_bgp(self):
        bgp_as = get(self._switch_data, "bgp_as")
        if bgp_as is None:
            return

        bgp_defaults = [bgp_default for bgp_default in get(self._switch_data, 'bgp_defaults')]
        if get(self._hostvars, 'bgp_maximum_paths') is not None:
            max_paths_str = f"maximum-paths {str(get(self._hostvars, 'bgp_maximum_paths'))}"
            if get(self._hostvars, 'bgp_ecmp') is not None:
                max_paths_str += f" ecmp {str(get(self._hostvars, 'bgp_ecmp'))}")
            
            bgp_defaults.append(max_path_str)

        return {'as': get(self._switch_data, 'bgp_as'),
                     'router_id': get(self._switch_data, 'router_id'),
                     'bgp_defaults': bgp_defaults}
```

Reply:
`[bgp_default for bgp_default in get(self._switch_data, 'bgp_defaults')]` should have a default value in case `bgp_defaults` is not set. `[bgp_default for bgp_default in get(self._switch_data, 'bgp_defaults', [])]`